pex/pip/version.py

```diff
@@ -93,57 +149,67 @@ def values(cls):
         version="22.2.2",
         setuptools_version="65.3.0",
         wheel_version="0.37.1",
-        requires_python=">=3.7",
+        requires_python=">=3.7,<3.12",
     )
 
     v22_3 = PipVersionValue(
         version="22.3",
         setuptools_version="65.5.0",
         wheel_version="0.37.1",
-        requires_python=">=3.7",
+        requires_python=">=3.7,<3.12",
     )
 
     v22_3_1 = PipVersionValue(
         version="22.3.1",
         setuptools_version="65.5.1",
         wheel_version="0.37.1",
-        requires_python=">=3.7",
+        requires_python=">=3.7,<3.12",
     )
 
     v23_0 = PipVersionValue(
         version="23.0",
         setuptools_version="67.2.0",
         wheel_version="0.38.4",
-        requires_python=">=3.7",
+        requires_python=">=3.7,<3.12",
     )
 
     v23_0_1 = PipVersionValue(
         version="23.0.1",
         setuptools_version="67.4.0",
         wheel_version="0.38.4",
-        requires_python=">=3.7",
+        requires_python=">=3.7,<3.12",
     )
 
     v23_1 = PipVersionValue(
         version="23.1",
         setuptools_version="67.6.1",
         wheel_version="0.40.0",
-        requires_python=">=3.7",
+        requires_python=">=3.7,<3.12",
     )
 
     v23_1_1 = PipVersionValue(
         version="23.1.1",
         setuptools_version="67.7.1",
         wheel_version="0.40.0",
-        requires_python=">=3.7",
+        requires_python=">=3.7,<3.12",
     )
 
     v23_1_2 = PipVersionValue(
         version="23.1.2",
         setuptools_version="67.7.2",
         wheel_version="0.40.0",
+        requires_python=">=3.7,<3.12",
+    )
+
+    v23_2 = PipVersionValue(
+        version="23.2.dev0+8a1eea4a",
+        requirement="pip @ git+https://github.com/pypa/pip@8a1eea4aaedb1fb1c6b4c652cd0c43502f05ff37",
+        setuptools_version="67.8.0",
+        wheel_version="0.40.0",
         requires_python=">=3.7",
+        hidden=True,
```

Comment:
Some complication was added here to be able to use a Pip version I want to yank, but still uphold the guaranty of not breaking users. This flag and the attendant complication above all work to make it so a release of Pex between now and the moment it can be upgraded to a Pip that support Python 3.12 won't expose a version of Pip on the CLI that will later get yanked.